swarm/network/discovery.go

```diff
@@ -55,15 +55,19 @@ func (self *discPeer) NotifyPeer(p Peer, po uint8) error {
 	resp := &peersMsg{
 		Peers: []*peerAddr{&peerAddr{OAddr: p.OverlayAddr(), UAddr: p.UnderlayAddr()}}, // perhaps the PeerAddr interface is unnecessary generalization
 	}
-	return self.Send(resp)
+	go self.Send(resp)
```

Comment:
if we change p2p/protocols Send to async, no need for individual calls to be called in a goroutine. 
Please remove everywhere.